subsystem/kubeapi_test.go

```diff
@@ -1764,6 +1764,87 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		}, "1m", "10s").Should(BeTrue())
 	})
 
+	It("Move Agent to another cluster", func() {
+		By("Create first cluster")
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+		deployAgentClusterInstallCRD(ctx, kubeClient, aciSNOSpec, clusterDeploymentSpec.ClusterInstallRef.Name)
+
+		clusterKey := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKey, waitForReconcileTimeout)
+		configureLocalAgentClient(cluster.ID.String())
+		host := registerNode(ctx, *cluster.ID, "hostname1")
+		key := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      host.ID.String(),
+		}
+
+		By("Check Agent Event URL exists")
+		Eventually(func() string {
+			agent := getAgentCRD(ctx, kubeClient, key)
+			return agent.Status.DebugInfo.EventsURL
+		}, "30s", "10s").ShouldNot(Equal(""))
+		firstAgentEventsURL := getAgentCRD(ctx, kubeClient, key).Status.DebugInfo.EventsURL
+
+		By("Create New ClusterDeployment")
+		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		clusterDeploymentSpec2 := getDefaultClusterDeploymentSpec(secretRef)
+		aciSNOSpec2 := getDefaultSNOAgentClusterInstallSpec(clusterDeploymentSpec2.ClusterName)
+		infraEnvSpec2 := getDefaultInfraEnvSpec(secretRef, clusterDeploymentSpec2)
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec2)
+		deployInfraEnvCRD(ctx, kubeClient, "infraenv2", infraEnvSpec2)
+		// Create AgentClusterInstall without creating the clusterImageSet
```

Comment:
General comment, not for this PR specifically.
We should think if the call to `deployClusterImageSetCRD` in `deployAgentClusterInstallCRD` make sense.
Perhaps we should call `deployClusterImageSetCRD` in the test setup.